crates/e2e/src/prover/block.rs

```diff
@@ -1,10 +1,490 @@
+            .await
+            .expect("Failed to get blobs")
+            .unwrap_or_default();
+        println!("Got {} blobs for block: {}", blobs.len(), block_number);
+
+        let extended_header = celestia_client
+            .header_get_by_height(block_number)
+            .await
+            .expect("Failed to get extended header");
+        let namespace_data = celestia_client
+            .share_get_namespace_data(&extended_header, namespace)
+            .await
+            .expect("Failed to get namespace data");
+        let mut proofs: Vec<NamespaceProof> = Vec::new();
+        for row in namespace_data.rows {
+            proofs.push(row.proof);
+        }
+        println!("Got NamespaceProofs, total: {}", proofs.len());
+
+        let mut executor_inputs: Vec<EthClientExecutorInput> = Vec::new();
+        for blob in blobs.as_slice() {
+            let data = match SignedData::decode(blob.data.as_slice()) {
+                Ok(data) => data.data.unwrap(),
+                Err(_) => continue,
+            };
+            let height = data.metadata.unwrap().height;
+            println!("Got SignedData for EVM block {height}");
+
+            let client_executor_input =
+                generate_client_executor_input(config::EVM_RPC_URL, height, chain_spec.clone(), genesis.clone())
+                    .await
+                    .expect("Failed to generate client executor input");
+            executor_inputs.push(client_executor_input);
+        }
+
+        let mut stdin = SP1Stdin::new();
+        let input = BlockExecInput {
+            header_raw: serde_cbor::to_vec(&extended_header.header).expect("Failed to serialize header"),
+            dah: extended_header.dah,
+            blobs_raw: serde_cbor::to_vec(&blobs).expect("Failed to serialize blobs"),
+            pub_key: pub_key.clone(),
+            namespace,
+            proofs,
+            executor_inputs: executor_inputs.clone(),
+            trusted_height: current_trusted_height,
+            trusted_root: current_trusted_root,
+        };
+
+        stdin.write(&input);
+        println!("Generating proof for block: {block_number}, trusted height: {current_trusted_height}");
+        let outputs = client
+            .execute(&block_prover_elf, &stdin)
+            .run()
+            .expect("failed to generate proof")
+            .0;
+
+        let public_values: BlockExecOutput =
+            bincode::deserialize(outputs.as_slice()).expect("Failed to deserialize public values");
+
+        current_trusted_height = public_values.new_height;
+        current_trusted_root = public_values.new_state_root.into();
+        trusted_heights.push(public_values.new_height);
+        trusted_roots.push(public_values.new_state_root.into());
+    }
+
+    // now we can generate proofs in parallel
+    let mut handles: Vec<JoinHandle<()>> = Vec::new();
+    for block_number in start_height..(start_height + num_blocks) {
```

Comment:
so because you run up front in `execute` mode your populate two vectors of trusted heights and roots to then pass into each as inputs when running in `prove` mode?

is that the gist here?